[gdal] build failure

### Operating system

Manjaro Linux 22.1.0 Talos (6.1.31-1 kernel)

### Compiler

gcc 12.2.1 20230201

### Steps to reproduce the behavior

```Shell
vcpkg install gdal
vcpkg install
```


### Failure logs

vcpkg console output:
```
~  vcpkg install gdal                                                                                                               
Computing installation plan...
The following packages will be built and installed:
    gdal[core,curl,default-features,expat,geos,gif,hdf5,iconv,jpeg,lerc,libkml,libspatialite,libxml2,lzma,netcdf,openjpeg,openssl,pcre2,png,postgresql,qhull,recommended-features,sqlite3,webp,zstd]:x64-linux -> 3.7.0#2
Detecting compiler hash for triplet x64-linux...
Restored 0 package(s) from /home/kian/.cache/vcpkg/archives in 6.54 us. Use --debug to see more details.
Installing 1/1 gdal:x64-linux...
Building gdal[core,curl,default-features,expat,geos,gif,hdf5,iconv,jpeg,lerc,libkml,libspatialite,libxml2,lzma,netcdf,openjpeg,openssl,pcre2,png,postgresql,qhull,recommended-features,sqlite3,webp,zstd]:x64-linux...
-- Using cached OSGeo-gdal-v3.7.0.tar.gz.
-- Cleaning sources at /home/kian/.local/vcpkg/buildtrees/gdal/src/v3.7.0-9fd58c78bf.clean. Use --editable to skip cleaning for the packages you specify.
-- Extracting source /home/kian/.local/vcpkg/downloads/OSGeo-gdal-v3.7.0.tar.gz
-- Applying patch find-link-libraries.patch
-- Applying patch fix-gdal-target-interfaces.patch
-- Applying patch libkml.patch
-- Using source at /home/kian/.local/vcpkg/buildtrees/gdal/src/v3.7.0-9fd58c78bf.clean
-- Found external ninja('1.11.1').
-- Configuring x64-linux
-- Building x64-linux-dbg
CMake Error at scripts/cmake/vcpkg_execute_build_process.cmake:134 (message):
    Command failed: /usr/bin/cmake --build . --config Debug --target install -- -v -j5
    Working Directory: /home/kian/.local/vcpkg/buildtrees/gdal/x64-linux-dbg
    See logs for more information:
      /home/kian/.local/vcpkg/buildtrees/gdal/install-x64-linux-dbg-out.log

Call Stack (most recent call first):
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_build.cmake:74 (vcpkg_execute_build_process)
  installed/x64-linux/share/vcpkg-cmake/vcpkg_cmake_install.cmake:16 (vcpkg_cmake_build)
  ports/gdal/portfile.cmake:97 (vcpkg_cmake_install)
  scripts/ports.cmake:147 (include)
```
[install-x64-linux-dbg-out.log](https://github.com/microsoft/vcpkg/files/11643382/install-x64-linux-dbg-out.log)


### Additional context

for manifest mode i was installing mapnik with `input-gdal` feature in `x64-linux-dynamic` triplet and i got the same errors.

> ~~~
> /usr/include/absl/base/policy_checks.h:79:2: error: #error "C++ versions less than C++14 are not supported."

This is unexpected use of system libraries. Neither abseil nor flatbuffers are in the dependencies of port gdal. I would like to see the config logs to see how it became part of the configuration.

--- a/ports/gdal/portfile.cmake
+++ b/ports/gdal/portfile.cmake
@@ -12,6 +12,9 @@ vcpkg_from_github(
 # `vcpkg clean` stumbles over one subdir
 file(REMOVE_RECURSE "${SOURCE_PATH}/autotest")
 
+# Avoid abseil, no matter if vcpkg or system
+vcpkg_replace_string("${SOURCE_PATH}/ogr/ogrsf_frmts/flatgeobuf/flatbuffers/base.h" [[__has_include("absl/strings/string_view.h")]] "(0)")
+
 # Cf. cmake/helpers/CheckDependentLibraries.cmake
 # The default for all `GDAL_USE_<PKG>` dependencies is `OFF`.
 # Here, we explicitly control dependencies provided via vpcpkg.

--- a/ports/gdal/vcpkg.json
+++ b/ports/gdal/vcpkg.json
@@ -1,7 +1,7 @@
 {
   "name": "gdal",
   "version-semver": "3.7.0",
-  "port-version": 2,
+  "port-version": 3,
   "description": "The Geographic Data Abstraction Library for reading and writing geospatial raster and vector data",
   "homepage": "https://gdal.org",
   "license": null,

--- a/versions/baseline.json
+++ b/versions/baseline.json
@@ -2722,7 +2722,7 @@
     },
     "gdal": {
       "baseline": "3.7.0",
-      "port-version": 2
+      "port-version": 3
     },
     "gdcm": {
       "baseline": "3.0.22",

--- a/versions/g-/gdal.json
+++ b/versions/g-/gdal.json
@@ -1,6 +1,11 @@
 {
   "versions": [
     {
+      "git-tree": "35cbea48f44c5d0836cd0b6f171396eae2745177",
+      "version-semver": "3.7.0",
+      "port-version": 3
+    },
+    {
       "git-tree": "f287b8dfcfba6e4f2d8ee8cdba0e63f0ae326fd7",
       "version-semver": "3.7.0",
       "port-version": 2

